src/app/components/dataTables/companies/companies.component.ts

```diff
@@ -100,7 +124,45 @@ export class CompaniesComponent implements OnInit {
     this.companyService
       .delete(data.id)
       .subscribe(response => {
-        this.loadData(this.pageSize, this.pageIndex, this.filter, this.sortColumn, this.sortDirection, this.searchAbleColumns);
+        this.loadData(this.pageSize, this.pageIndex, this.searchTerm, this.sortColumn, this.sortDirection, this.searchAbleColumns);
       });
   }
+
+  private setSubscriptions(): void{
+    this.setSearchTermSubscription();
+    this.setDisplayedColumnsSubscription();
+    this.setSearchedColumnsSubscription();
+    this.setHasChangedSubscription();
+  }
+
+  private setSearchTermSubscription(): void{
+    this.filterService.searchTerm.pipe(
+      debounceTime(500),
+      distinctUntilChanged()).subscribe(searchTerm => this.setSearchTerm(searchTerm));
+  }
+
+  private setDisplayedColumnsSubscription(): void{
+    this.filterService.displayAbleColumns.subscribe(displayedCol => {
+      this.displayedColumns = displayedCol;
+    });
+  }
+
+  private setSearchedColumnsSubscription(): void{
+    this.filterService.searchAbleColumns.subscribe(searchedCol => {
+      this.searchAbleColumns = searchedCol;
+    });
+  }
+
+  private setHasChangedSubscription(): void{
+    this.filterService.hasChanged.subscribe(() =>  this.reloadData());
+  }
+
+  private setSearchTerm(term: string): void{
+    this.searchTerm = term;
+    this.reloadData();
+  }
+
+  private reloadData(): void{
+    this.loadData(this.pageSize, this.pageIndex, this.searchTerm, this.sortColumn, this.sortDirection, this.searchAbleColumns);
+  }
```

Comment:
incearca sa creezi o componenta de baza, pe care o extinzi daca ai nevoie de lucruri in plus